mobile/lib/modules/home/views/home_page.dart

```diff
@@ -335,31 +335,41 @@ class HomePage extends HookConsumerWidget {
         child: Stack(
           children: [
             ref.watch(assetsProvider(currentUser?.isarId)).when(
-                  data: (data) => data.isEmpty
-                      ? buildLoadingIndicator()
-                      : ImmichAssetGrid(
-                          renderList: data,
-                          listener: selectionListener,
-                          selectionActive: selectionEnabledHook.value,
-                          onRefresh: refreshAssets,
-                          topWidget: const MemoryLane(),
-                        ),
-                  error: (error, _) => Center(child: Text(error.toString())),
-                  loading: buildLoadingIndicator,
-                ),
+              data: (data) =>
+              data.isEmpty
+                  ? buildLoadingIndicator()
+                  : ImmichAssetGrid(
+                renderList: data,
+                listener: selectionListener,
+                preselectedAssets: selection.value,
+                selectionActive: selectionEnabledHook.value,
+                onRefresh: refreshAssets,
+                topWidget: const MemoryLane(),
+              ),
+              error: (error, _) => Center(child: Text(error.toString())),
+              loading: buildLoadingIndicator,
+            ),
             if (selectionEnabledHook.value)
-              ControlBottomAppBar(
-                onShare: onShareAssets,
-                onFavorite: onFavoriteAssets,
-                onArchive: onArchiveAsset,
-                onDelete: onDelete,
-                onAddToAlbum: onAddToAlbum,
-                albums: albums,
-                sharedAlbums: sharedAlbums,
-                onCreateNewAlbum: onCreateNewAlbum,
-                onUpload: onUpload,
-                enabled: !processing.value,
-                selectionAssetState: selectionAssetState.value,
+              ref.watch(assetsProvider(currentUser?.isarId)).when(
+                data: (data) =>
+                data.isEmpty
+                    ? buildLoadingIndicator() : ControlBottomAppBar(
+                  listener: selectionListener,
+                  allAssets: data.loadAssets(0, data.totalAssets),
```

Comment:
Can't do that. It'll take forever for large collections - we explicitly made the views load assets in chunks (only those that are viewed or near the viewport).
So, to build a "select all" functionality, a bit more is involved. For mid-sized libraries it could be okay to only load all assets on demand (once you click that select all button). For large collections that'll still kill the app due to out of memory. You'd need to come up with a more complex solution for this to work properly.

Reply:
Ah ok thank you :)